toolchain/check/impl_lookup.cpp

```diff
@@ -577,13 +585,18 @@ auto LookupImplWitness(Context& context, SemIR::LocId loc_id,
     return SemIR::InstBlockIdOrError::MakeError();
   }
 
-  bool has_other_requirements = false;
-  auto interfaces = GetInterfacesFromConstantId(
-      context, query_facet_type_const_id, has_other_requirements);
-  if (has_other_requirements) {
+  auto [interfaces, special_requirements_mask] =
+      GetInterfacesFromConstantId(context, query_facet_type_const_id);
+  if (special_requirements_mask.has(
+          SemIR::FacetTypeInfo::SpecialRequirement::Other)) {
     // TODO: Remove this when other requirements go away.
     return SemIR::InstBlockId::None;
   }
+  if (special_requirements_mask.has(
+          SemIR::FacetTypeInfo::SpecialRequirement::TypeCanAggregateDestroy) &&
+      !TypeCanAggregateDestroy(context, query_self_const_id)) {
+    return SemIR::InstBlockId::None;
+  }
```

Comment:
Can we CHECK that if the facet type special requirements (not counting `other_requirements`, see below) then it has nothing else, no interfaces/constraints/etc, and then we can always return an answer directly here, either None or Empty?